### PR TITLE
test: add core WeakSet builtin unit tests

### PR DESCRIPTION
**Pull request by mrhapile**
This PR introduces initial unit tests for the WeakSet builtin.

Currently the WeakSet implementation relies entirely on test262 coverage
and does not have internal Rust-level tests in the Boa engine.

This PR adds:

core/engine/src/builtins/weak_set/tests.rs

The following behaviors are tested:

- WeakSet.prototype.add basic functionality
- WeakSet.prototype.has
- WeakSet.prototype.delete
- add() chaining behavior

These tests follow the same structure used by other builtin tests
such as WeakMap, Map, and Set.

Screenshot 2026-03-07 at 7 02 07 AM

Additional edge case tests will be added in a follow-up PR.

### CODE REVIEW

File: core/engine/src/builtins/weak_set/mod.rs [hunk 1]
```diff
@@ -253,3 +253,6 @@ impl WeakSet {
         Ok(set.contains_key(value.inner()).into())
     }
 }
+
+#[cfg(test)]
+mod tests;
```
[the remark this answers is not in the document]
Comment (mrhapile) [reply core/engine/src/builtins/weak_set/mod.rs]: implemented in latest commit!

**Comment by jedel1043** [comment]
Superseeded by #4914
